cli/azd/pkg/project/container_helper.go

```diff
@@ -216,6 +221,52 @@ func (ch *ContainerHelper) Deploy(
 ) *async.TaskWithProgress[*ServiceDeployResult, ServiceProgress] {
 	return async.RunTaskWithProgress(
 		func(task *async.TaskContextWithProgress[*ServiceDeployResult, ServiceProgress]) {
+			if dotnetDetails, ok := packageOutput.Details.(*dotNetSdkPublishConfiguration); ok {
+				task.SetProgress(NewServiceProgress("Logging into container registry"))
+				dockerCreds, err := ch.Credentials(ctx, serviceConfig, targetResource)
```

Comment:
Guessing the shape of this is going to change with your upcoming changes, @vhvb1989.  Guessing we will get to a place where I can just call `ch.Login(...)` (which may no longer require docker?) and then `ch.dotNetCli.PublishContainer`?

Anyway, calling your eye to this as a sense of what I'd like to be able to do.